### ISSUE TITLE
[Bug] TH Directory selector locks out window controls on Linux

### ISSUE THREAD

**Issue by lewri** [CorsixTH/CorsixTH#2711]
### Describe the issue

When you launch CorsixTH for the first time the mouse isn't locked to the window but the `minimize`, `maximize` and `close` buttons on the top of the window aren't clickable.

You can press the Exit button on the directory tree window still and it will close.

### Steps to Reproduce

(make sure capture mouse is set to true in confih)
1. Install CorsixTH on Linux without having a common TH location
2. Open game
3. Notice how the buttons don't work.

### Expected Behaviour

The buttons on the window should be responding. It could be a Linux issue.

### Have you enabled any current CorsixTH Experimental Features?

- [ ] Right mouse panning
- [ ] Falling actions

### Save Game

Save game attached to report: Yes/No

### CorsixTH Version

0.67 (haven't checked 0.68.0 yet but nothing has changed there afaik)

### Operating System

Linux Mint 22

### Theme Hospital Version

GOG.com

### Additional Information

![image](https://github.com/user-attachments/assets/adf7a2a1-54d3-40d6-8759-5cc0c049bf47)

Have also tried the Flatpak version on Vanilla OS 2

**Comment by lewri** [comment]
Correction -- the Flatpak version would have been 0.68.0

**Comment by TheCycoONE** [comment]
The window manager matters here, and X11 or wayland.

**Comment by TheCycoONE** [comment]
I cannot reproduce with 0.68 on ArchLinux in KDE (Wayland)

**Comment by lewri** [comment]
@TheCycoONE so it looks like the issue here is because on this specific case we're not providing a cursor to be the `capture_mouse` target (more or less, some thought process may be wrong).

I've hacked in a temporary cursor to the game in Linux and it then behaves as it should.
![image](https://github.com/user-attachments/assets/676a848f-bea3-4238-8cfe-2935c2fe29dd)

**Comment by Alberth289346** [comment]
Yes, you can select the directories in the purple area in the application canvas, but the original issue was reaching the window control buttons at the top-right (see also OP).

I reproduced not being able to reach them. That does make sense to me, since locking the cursor in a window means confining it the area of the application. The edge and the top bar with the controls around that area are  NOT considered application area in Unix.

In other words, the OP works as expected in Linux for me.

**Comment by TheCycoONE** [comment]
Only an issue when the system doesn't allow capture of the hardware cursor but does allow blocking events outside of the window area I suppose.
